### ISSUE TITLE
Performance regression on closing statement with Hikari

### ISSUE THREAD

**Issue by NathanEckert** [aws/amazon-redshift-jdbc-driver#100]
## Driver version

Upgrade from driver `2.1.0.19` to `2.1.0.20`

## Redshift version

`PostgreSQL 8.0.2 on i686-pc-linux-gnu, compiled by GCC gcc (GCC) 3.4.2 20041017 (Red Hat 3.4.2-6.fc3), Redshift 1.0.57864`

## Client Operating System

`Linux Ubuntu 22.04.3 LTS (Jammy Jellyfish)`

## JAVA/JVM version

```
Apache Maven 3.8.6 (84538c9988a25aec085021c365c560670ad80f63)
Maven home: /home/nec/.sdkman/candidates/maven/current
Java version: 11.0.20, vendor: Eclipse Adoptium, runtime: /home/nec/.sdkman/candidates/java/11.0.20-tem
Default locale: en_US, platform encoding: UTF-8
OS name: "linux", version: "6.2.0-35-generic", arch: "amd64", family: "unix"
```

Additional dependency:

```xml
<dependency>
<groupId>com.zaxxer</groupId>
<artifactId>HikariCP</artifactId>
<version>5.0.1</version>
</dependency>
```

## Problem description

When upgrading driver version, a performance regression is observed when using the driver in conjunction with Hikari.

Before the upgrade, it took 13 ms to close the statement, versus more than 2 seconds after the upgrade.

This issue is only observed with Hikari.

## Reproduction code

```java
package com.activeviam.database.redshift;

import com.activeviam.directquery.test.internal.redshift.RedshiftTestConstants;
import com.amazon.redshift.RedshiftProperty;
import com.zaxxer.hikari.HikariConfig;
import com.zaxxer.hikari.HikariDataSource;
import java.sql.Connection;
import java.sql.PreparedStatement;
import java.sql.SQLException;
import java.util.Properties;
import java.util.concurrent.TimeUnit;
import lombok.Cleanup;
import org.junit.jupiter.api.Test;

public class TestPerformanceJDBCDriver {

	private static final String CONNECTION_STRING = RedshiftTestConstants.REDSHIFT_CLUSTER_URL;
	private static final String QUERY = "SELECT 1;";


	@Test
	void testPreparedStatement() throws SQLException {

		Connection connection = null;
		PreparedStatement statement = null;

		@Cleanup
		final HikariDataSource dataSource = createHikariDatasource(CONNECTION_STRING + RedshiftTestConstants.TEST_DATABASE_NAME);

		try {
			connection = dataSource.getConnection();
			statement = connection.prepareStatement(QUERY);
			statement.execute();
		} catch (SQLException e) {
			throw new RuntimeException(e);
		} finally {

			if (statement != null) {
				final long startClosingStatement = System.nanoTime();
				statement.close();
				System.out.println("Closing statement took: " + TimeUnit.NANOSECONDS.toMillis(System.nanoTime() - startClosingStatement) + " ms.");
			}
			if (connection != null) {
				connection.close();
			}

		}


	}

	private static HikariDataSource createHikariDatasource(final String connectionString) {
		final Properties jdbcProperties = new Properties();
		RedshiftProperty.USER.set(jdbcProperties, RedshiftTestConstants.getUsername());
		RedshiftProperty.PASSWORD.set(jdbcProperties, RedshiftTestConstants.getRedshiftPassword());

		final HikariConfig config = new HikariConfig();
		config.setDataSourceProperties(jdbcProperties);
		config.setJdbcUrl(connectionString);
		return new HikariDataSource(config);
	}
}

```

The previous outputs `Closing statement took: 2021 ms.` with driver version 2.1.0.20, and `Closing statement took: 14 ms.` with driver version 2.1.0.19

**Comment by bhvkshah** [comment]
@NathanEckert we are aware of this issue and have a fix in place. It should available as part of the next release, version 2.1.0.22, slated for mid-November

**Comment by bhvkshah** [comment]
Hi @NathanEckert , driver version 2.1.0.22 has been released which contains the fix for this issue. Thank you again for your contribution to the Redshift JDBC Driver!
